### PR TITLE
Initial Project Setup

### PR DESCRIPTION
**Pull request by bauer-andreas**
# Initial Project Setup

## :recycle: Current situation & Problem
This PR marks the start of the SpeziNetworking package. It includes the `ByteCoding` target previously located in `SpeziFileFormats`. Together with UInt24 support previously part of `ByteCoding` and the `MedFloat16` implementation currently located in `BluetoothServices` of `SpeziBluetooth`, it forms the new `SpeziNumerics` target for networking related numeric types or type support.

## :gear: Release Notes 
* Add `ByteCoding` target.
* Add `SpeziNumerics` target with 24 bit Integer support and MedFloat16 implementation.
* Initialized repo with updated name, badges and documentation.

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/build-and-test.yml
Comment (PSchmiedmayer) [inline .github/workflows/build-and-test.yml:1]: Nice; thanks for setting this up!

File: CITATION.cff
Comment (PSchmiedmayer) [inline CITATION.cff:1]: Would be great to update the people here and add yourself and everyone that contributed to the relevant code as done in the CONTRIBUTORS file.

File: README.md [hunk 1]
```diff
@@ -1,55 +1,49 @@
...
+[![codecov](https://codecov.io/gh/StanfordSpezi/SpeziNetworking/graph/badge.svg?token=emNLUokqWO)](https://codecov.io/gh/StanfordSpezi/SpeziNetworking)
+[![](https://img.shields.io/endpoint?url=https%3A%2F%2Fswiftpackageindex.com%2Fapi%2Fpackages%2FStanfordSpezi%2FSpeziNetworking%2Fbadge%3Ftype%3Dswift-versions)](https://swiftpackageindex.com/StanfordSpezi/SpeziNetworking)
+[![](https://img.shields.io/endpoint?url=https%3A%2F%2Fswiftpackageindex.com%2Fapi%2Fpackages%2FStanfordSpezi%2FSpeziNetworking%2Fbadge%3Ftype%3Dplatforms)](https://swiftpackageindex.com/StanfordSpezi/SpeziNetworking)
 
 
-## How To Use This Template
+A collection of networking-related infrastructure to support Spezi applications.
 
-The template repository contains a template Swift Package, including a continuous integration setup. 
+## Overview
```
Comment (PSchmiedmayer) [inline README.md:21]: Would be nice if we can either already link to some documentation (given that we know what the links will be) or create an issue to do this in a separate PR.

Would be good to get a simple list of types + one liner, and link to docs to show what it does here in a bullet point list to give people visiting this package a first glimpse what to expect here.

**Review by PSchmiedmayer — APPROVED** [review]
Amazing job @Supereg; this is growing so fast 🚀

File: README.md (hunk 1 above)
Comment (bauer-andreas) [reply README.md:21]: Tracked it as a separate issue. Waiting for SPI to build the documentation 👍

File: CITATION.cff
Comment (bauer-andreas) [reply CITATION.cff:1]: Automerge kicked in before I could edit these. Will do in the follow up PR 👍

**Pull request merged**